<launch>
    <include file="$(find husky_gazebo)/launch/husky_empty_world.launch" />

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find coffeebot)/config/cbrc_plan_test.rviz" output="screen"/>

    <!-- Run the map server -->
    <arg name="map_file" default="$(find coffeebot)/config/floor3_map.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
        <remap from="/map"       to="/floor_map" />
    </node>

    <!-- Run AMCL -->
    <include file="$(find husky_navigation)/launch/amcl.launch" />

    <!--- Run Move Base -->
    <arg name="no_static_map" default="false"/>

    <!--arg name="base_global_planner" default="navfn/NavfnROS"/-->
    <arg name="base_global_planner" default="astar_cb_global_planner/AStar_CB_GlobalPlanner"/>
    <!--arg name="base_global_planner" default="global_planner/GlobalPlanner"/-->
    <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
    <!-- <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/> -->
    <node pkg="tf" type="static_transform_publisher"
		 name="map_tf_publisher" args="0 0 0 0 0 0 /map /odom 10" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <param name="base_global_planner" value="$(arg base_global_planner)"/>
        <param name="base_local_planner" value="$(arg base_local_planner)"/>
        <rosparam file="$(find husky_navigation)/config/planner.yaml" command="load"/>
        <!--<rosparam file="$(find coffeebot)/config/planner.yaml" command="load"/>-->

        <!-- observation sources located in costmap_common.yaml -->
        <rosparam file="$(find coffeebot)/config/costmap_common.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find coffeebot)/config/costmap_common.yaml" command="load" ns="local_costmap" />

        <!-- local costmap, needs size -->
        <rosparam file="$(find husky_navigation)/config/costmap_local.yaml" command="load" ns="local_costmap" />
        <param name="local_costmap/width" value="10.0"/>
        <param name="local_costmap/height" value="10.0"/>

        <!-- static global costmap, static map provides size -->
        <rosparam file="$(find husky_navigation)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/>

        <!-- global costmap with laser, for odom_navigation_demo -->
        <rosparam file="$(find husky_navigation)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
        <!--param name="global_costmap/width" value="100.0" if="$(arg no_static_map)"/-->
        <!--param name="global_costmap/height" value="100.0" if="$(arg no_static_map)"/-->
        <remap from="/map" to="/floor_map" />
    </node>
</launch>
